tests/unit/test_version.py

```diff
@@ -0,0 +1,31 @@
+from version import get_snap_version
+from version import get_snap_revision
+from version import get_upstream_versions
+from unittest.mock import mock_open, patch
+from pathlib import Path
+import json
+
+
+TEST_VERSIONS = {"kube": "v1.21", "cni": "0.0.1"}
+
+
+@patch("version.environ", {"SNAP_VERSION": "v1.21.11"})
+def test_get_snap_version():
+    assert get_snap_version() == "v1.21.11"
+
+
+@patch("version.environ", {"SNAP_REVISION": "3086"})
+def test_get_snap_revision():
+    assert get_snap_revision() == "3086"
```

Comment:
I would probably avoid mocking for this, and instead just set the environment variable

Reply:
sure! Why though? Out of curiosity

Reply:
This way, the tests are too closely tied to the implementation (e.g. if the implementation were to use `os.getenv` instead, these tests would not work). 

Mocks are best suited for outputs/integration with external services. They also serve as a way of disconnecting the testing environment from the real one. This is useful when we want to mock the output of an external command, etc, but is probably a dis-service when checking environment (where no mocks are really needed, the test case itself is isolated).

Reply:
makes total sense! thanks for sharing your thoughts